float64 x
float64 y
float64 z
float32 confidence
float32[] embedding
